src/hooks.ts

```diff
@@ -18,3 +19,57 @@ export function useFlexSize() {
   const { size } = useContext(boxContext)
   return size
 }
+
+export function useNode() {
+  const { node } = useContext(boxContext)
+  return node
+}
+
+/**
+ * explicitly set the size of the box's yoga node
+ * @requires that the surrounding Flex-Element has `disableSizeRecalc` set to `true`
+ */
+export function useSetSize(): (width: number, height: number) => void {
+  const { requestReflow, scaleFactor } = useContext(flexContext)
+  const node = useNode()
+
+  const sync = useCallback(
+    (width: number, height: number) => {
+      if (node == null) {
+        throw new Error('yoga node is null. sync size is impossible')
+      }
+      node.setWidth(width * scaleFactor)
+      node.setHeight(height * scaleFactor)
+      requestReflow()
+    },
+    [node, requestReflow]
+  )
+
+  return sync
+}
+
+const helperVector = new Vector3()
+
+/**
+ * explicitly sync the yoga node size with a mesh's geometry and uniform global scale
+ * @requires that the surrounding Flex-Element has `disableSizeRecalc` set to `true`
+ */
+export function useSyncGeometrySize(): (mesh: Mesh) => void {
```

Comment:
Same thing here for the function signature

Reply:
That would destroy the use case I described in the PR description or would require some sort of state and a rerender when the geometry is changed.

Currently, I can pass this function to `onSync` of the r3f/drei Text element and it can sync the box's size asynchronously.

Reply:
Sorry I'm dumb, that's the return value that is a function 😭